hazelcast/src/main/java/com/hazelcast/nio/VersionAware.java

```diff
@@ -16,16 +16,14 @@
 
 package com.hazelcast.nio;
 
-import com.hazelcast.version.Version;
-
 /**
  * Enables getting the version from the implementing object.
  * It may be any Version (cluster version, node version, custom version, etc.) It's up to the implementer.
  */
 public interface VersionAware {
 
     /**
-     * @return the version or Version.UNKNOWN if version is unknown to the object.
+     * @return the version or {@code null} if version is unknown to the object.
      */
```

Comment:
It will require the users to do a null check in the conditional-deserialzation. I don't know what's better though...